impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -0,0 +1,262 @@
+     * @throws InvalidKeyException
+     */
+    PrivateKey parsePrivateKey(String privateKeyFilePath) throws IOException, InvalidKeyException {
+        Assert.notNull(privateKeyFilePath, "privateKeyFilePath may not be null");
+
+        File privateKeyFile = new File(privateKeyFilePath);
+        if (!privateKeyFile.exists()) {
+            throw new IllegalArgumentException("privateKeyFile " + privateKeyFile + " does not exist");
+        }
+
+        Reader reader = new InputStreamReader(new FileInputStream(privateKeyFilePath), "UTF-8");
+
+        PrivateKey privateKey = getPrivateKeyFromPEM(reader);
+        String algorithm = privateKey.getAlgorithm();
+
+        if (!algorithm.equals("RSA") &&
+            !algorithm.equals("EC")) {
+            throw new InvalidKeyException("Supplied privateKey is not an RSA or EC key - " + algorithm);
+        }
+
+        return privateKey;
+    }
+
+    /**
+     * Get Private key from input PEM file.
+     *
+     * @param reader
+     * @return {@link PrivateKey}
+     * @throws IOException
+     */
+    PrivateKey getPrivateKeyFromPEM(Reader reader) throws IOException {
+        PrivateKey privateKey;
+
+        try (PEMParser pemParser = new PEMParser(reader)) {
+            JcaPEMKeyConverter jcaPEMKeyConverter = new JcaPEMKeyConverter();
+            Object pemContent = pemParser.readObject();
+
+            if (pemContent instanceof PEMKeyPair) {
+                PEMKeyPair pemKeyPair = (PEMKeyPair) pemContent;
+                KeyPair keyPair = jcaPEMKeyConverter.getKeyPair(pemKeyPair);
+                privateKey = keyPair.getPrivate();
+            } else if (pemContent instanceof PrivateKeyInfo) {
+                PrivateKeyInfo privateKeyInfo = (PrivateKeyInfo) pemContent;
+                privateKey = jcaPEMKeyConverter.getPrivateKey(privateKeyInfo);
+            } else {
+                throw new IllegalArgumentException("Unsupported Private Key format '" +
+                    pemContent.getClass().getSimpleName() + '"');
+            }
+        }
+
+        return privateKey;
+    }
+
+    /**
+     * Create token client config from the supplied API client config.
+     *
+     * Token client needs to retry http 401 errors only once which is not the case with the API client.
+     * We therefore effect this token client specific config by setting 'retryMaxElapsed'
+     * & 'retryMaxAttempts' fields.
+     *
+     * @param apiClientConfiguration
+     * @return ClientConfiguration to be used by token retrieval client.
+     */
+    ClientConfiguration constructTokenClientConfig(ClientConfiguration apiClientConfiguration) {
+        ClientConfiguration tokenClientConfiguration = new ClientConfiguration();
+
+        if (apiClientConfiguration.getClientCredentialsResolver() != null)
+            tokenClientConfiguration.setClientCredentialsResolver(apiClientConfiguration.getClientCredentialsResolver());
+
+        if (apiClientConfiguration.isCacheManagerEnabled())
```

Comment:
You probably always want to use the disabled Cache manager for the token client

Reply:
done